src/app/utils/field-validation/validators/childrenValidator.ts

```diff
@@ -0,0 +1,128 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import {
+  ChildrenCompoundFieldBase,
+  MyInfoChildAttributes,
+} from '../../../../../shared/types'
+import {
+  IChildrenCompoundFieldSchema,
+  OmitUnusedValidatorProps,
+} from '../../../../types/field'
+import { ResponseValidator } from '../../../../types/field/utils/validation'
+import { ProcessedChildrenResponse } from '../../../modules/submission/submission.types'
+
+type ChildrenValidator = ResponseValidator<ProcessedChildrenResponse>
+type ChildrenValidatorConstructor = (
+  checkboxField: OmitUnusedValidatorProps<IChildrenCompoundFieldSchema>,
+) => ChildrenValidator
+
+/**
+ * Returns a validator to check if answerArray is empty
+ */
+const childrenAnswerValidator: ChildrenValidator = (response) => {
+  const { answerArray } = response
+
+  return answerArray.length === 0
+    ? left(`CheckboxValidator:\t Answer is empty array`)
```

Comment:
`CheckboxValidator:` ? 😅 

(same comment on all the error blocks below)

Reply:
Ah you caught where I copied the code from :).